tests/microos/network_bonding.pm

```diff
@@ -11,45 +11,157 @@ use strict;
 use warnings;
 use testapi;
 use power_action_utils "power_action";
+use serial_terminal qw(select_serial_terminal);
+
+sub check_connectivity {
+    my ($bond_name) = @_;
+    my $ping_host = "conncheck.opensuse.org";
+    my $ping_command = "ping -c1 -I $bond_name $ping_host";
+    assert_script_run $ping_command;
+}
+
+sub get_nics {
+    my ($bond_name) = @_;
+    my @devices;
+
+    foreach my $line (split('\n', script_output('nmcli -g DEVICE conn show'))) {
+        # Skip empty lines
+        next if $line =~ /^\s*$/;
+
+        # Skip the loopback device and the bond interface
+        next if $line eq 'lo';
+        next if $line eq $bond_name;
+
+        # Add valid device names to the array
+        push @devices, $line;
+    }
+
+    return @devices;
+}
+
+sub delete_existing_connections {
+    my $output = script_output('nmcli -g DEVICE,UUID conn show');
+    my %seen_uuids;    # Hash to keep track of processed UUIDs
+
+    foreach my $line (split "\n", $output) {
+        # Skip empty lines
+        next if $line =~ /^\s*$/;
+
+        my ($device, $uuid) = split /:/, $line;
+
+        # Skip loopback device
+        next if defined $device && $device eq 'lo';
+
+        # Check if the UUID has been seen before, then skip to the next iteration
+        next if $seen_uuids{$uuid};
```

Comment:
Checking existence of the key makes sense over here before checking whether any value is set

Reply:
Changed